src/helics/shared_api_library/helicsCallbacks.h

```diff
@@ -140,6 +140,28 @@ HELICS_EXPORT void helicsFederateSetTimeUpdateCallback(HelicsFederate fed,
                                                        void* userdata,
                                                        HelicsError* err);
 
+/**
+ * Set callback for the federate mode change.
+ *
+ * @details This callback will be executed every time the operating mode of the federate changes.
+ *
+ * @param fed The federate to set the callback for.
+ * @param stateChange A callback with signature void(HelicsFederateState newState, HelicsFederateState oldState, void *userdata);
+ *                 The function arguments are the new time value, a bool indicating that the time is iterating, and pointer to the userdata.
```

Comment:
``suggestion
 *                 The function arguments are the new state, the old state, and pointer to the userdata.
```